### PR TITLE
Fix Scope.exists and some missing Inlined trees

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Rebased

File: compiler/src/dotty/tools/dotc/transform/YCheckPositions.scala [hunk 1]
```diff
@@ -57,8 +57,9 @@ class YCheckPositions extends Phases.Phase {
   }
 
   private def isMacro(call: Tree)(implicit ctx: Context) = {
-    if (ctx.phase <= ctx.typerPhase.next) call.symbol.is(Macro)
-    else (call.symbol.unforcedDecls.exists(_.is(Macro)) || call.symbol.unforcedDecls.toList.exists(_.is(Macro)))
+    if (ctx.phase <= ctx.postTyperPhase) call.symbol.is(Macro)
+    else call.isInstanceOf[Select] // The call of a macro after typer is encoded as a Select while other inlines are Ident
+                                   // TODO remove this distinction once Inline nodes of expanded macros can be trusted (also in Inliner.inlineCallTrace)
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/transform/YCheckPositions.scala]: Looking at the method alone, `call.isInstanceOf[Select]` seems to be a too weak test.

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/YCheckPositions.scala]: This is a heuristic to know if we should `YCheck` the positions inside this `Inlined` tree or not.

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/YCheckPositions.scala]: At some point we will test the contents of these `Inlined` trees.  Then this test would be removed.

**Pull request merged**